cpp/log/etcd_consistent_store-inl.h

```diff
@@ -238,55 +260,31 @@ bool LessBySequenceNumber(const EntryHandle<Logged>& lhs,
 
 
 template <class Logged>
-util::Status EtcdConsistentStore<Logged>::GetSequencedEntries(
-    std::vector<EntryHandle<Logged>>* entries) const {
-  util::Status status(GetAllEntriesInDir(GetFullPath(kSequencedDir), entries));
-  sort(entries->begin(), entries->end(), LessBySequenceNumber<Logged>);
-  if (status.ok()) {
-    for (const auto& entry : *entries) {
-      CHECK(entry.Entry().has_sequence_number());
-    }
+util::Status EtcdConsistentStore<Logged>::GetSequenceMapping(
+    EntryHandle<ct::SequenceMapping>* sequence_mapping) const {
+  util::Status status(GetEntry(GetFullPath(kSequenceFile), sequence_mapping));
+  if (!status.ok()) {
+    return status;
   }
-  return status;
-}
-
-
-template <class Logged>
-util::Status EtcdConsistentStore<Logged>::GetSequencedEntry(
-    const int64_t sequence_number, EntryHandle<Logged>* entry) const {
-  CHECK_GE(sequence_number, 0);
-  util::Status status(GetEntry(GetSequencedPath(sequence_number), entry));
-  if (status.ok()) {
-    CHECK(entry->Entry().has_sequence_number());
+  CheckMappingIsOrdered(sequence_mapping->Entry());
+  if (serving_sth_) {
+    CheckMappingIsContiguousWithServingTree(serving_sth_->Entry(),
+                                            sequence_mapping->Entry());
   }
-
-  return status;
+  return util::Status::OK;
 }
 
 
 template <class Logged>
-util::Status EtcdConsistentStore<Logged>::AssignSequenceNumber(
-    const int64_t sequence_number, EntryHandle<Logged>* entry) {
-  CHECK_GE(sequence_number, 0);
-  CHECK(!entry->Entry().has_sequence_number());
-  if (entry->Entry().has_provisional_sequence_number()) {
-    CHECK_EQ(sequence_number, entry->Entry().provisional_sequence_number());
-  } else {
-    entry->MutableEntry()->set_provisional_sequence_number(sequence_number);
-  }
-  // Record provisional sequence number:
-  util::Status status(UpdateEntry(entry));
-  if (!status.ok()) {
-    return status;
+util::Status EtcdConsistentStore<Logged>::UpdateSequenceMapping(
+    EntryHandle<ct::SequenceMapping>* entry) {
+  CHECK(entry->HasHandle());
+  CheckMappingIsOrdered(entry->Entry());
+  if (serving_sth_) {
```

Comment:
I've moved that `CheckMappingIsContiguousWithServingTree()` method into `EtcdConsistentStore` and grab the lock locally now.